Wrappers/Python/test/test_CIL_CVXPy.py

```diff
@@ -0,0 +1,148 @@
+
+    def setUp(self):
+
+        # default test data
+        self.data = dataexample.CAMERA.get(size=(32, 32))
+
+    
+    # Create gradient operator as sparse matrix that will be used in CVXpy
+    def sparse_gradient_matrix(self, shape, direction='forward', order=1, boundaries='Neumann', **kwargs):
+        
+        len_shape = len(shape)    
+        allMat = dict.fromkeys(range(len_shape))
+        discretization = kwargs.get('discretization',[1.0]*len_shape)
+
+        if order == 1:
+
+            for i in range(0,len_shape):
+
+                if direction == 'forward':
+
+                    mat = 1/discretization[i] * sp.spdiags(np.vstack([-np.ones((1,shape[i])),np.ones((1,shape[i]))]), [0,1], shape[i], shape[i], format = 'lil')
+
+                    if boundaries == 'Neumann':
+                        mat[-1,:] = 0
+                    elif boundaries == 'Periodic':
+                        mat[-1,0] = 1
+
+                elif direction == 'backward':
+
+                    mat = 1/discretization[i] * sp.spdiags(np.vstack([-np.ones((1,shape[i])),np.ones((1,shape[i]))]), [-1,0], shape[i], shape[i], format = 'lil')
+
+                    if boundaries == 'Neumann':
+                        mat[:,-1] = 0
+                    elif boundaries == 'Periodic':
+                        mat[0,-1] = -1
+
+                tmpGrad = mat if i == 0 else sp.eye(shape[0])
+
+                for j in range(1, len_shape):
+
+                    tmpGrad = sp.kron(mat, tmpGrad ) if j == i else sp.kron(sp.eye(shape[j]), tmpGrad )
+
+                allMat[i] = tmpGrad
+
+        else:
+            raise NotImplementedError    
+
+        return allMat        
+
+    def tv_cvxpy_regulariser(self, u, isotropic=True, direction = "forward", boundaries = "Neumann"):
+
+        G = self.sparse_gradient_matrix(u.shape, direction = direction, order = 1, boundaries = boundaries)   
+
+        DX, DY = G[1], G[0]
+    
+        if isotropic:
+            return cp.sum(cp.norm(cp.vstack([DX @ cp.vec(u), DY @ cp.vec(u)]), 2, axis = 0))
+        else:
+            return cp.sum(cp.norm(cp.vstack([DX @ cp.vec(u), DY @ cp.vec(u)]), 1, axis = 0)) 
+
+    def test_cil_vs_cvxpy_totalvariation_isotropic(self):
+
+        # solution
+        u_cvx = cp.Variable(self.data.shape)
+
+        # regularisation parameter
+        alpha = 0.1
+
+        # fidelity term
+        fidelity = 0.5 * cp.sum_squares(u_cvx - self.data.array)   
+        regulariser = alpha * self.tv_cvxpy_regulariser(u_cvx)
+
+        # objective
+        obj =  cp.Minimize( regulariser +  fidelity)
+        prob = cp.Problem(obj, constraints = [])
+
+        # Choose solver ( SCS, MOSEK(license needed) )
+        tv_cvxpy = prob.solve(verbose = True, solver = cp.SCS)        
+
+        # use TotalVariation from CIL (with Fast Gradient Projection algorithm)
+        TV = alpha * TotalVariation(max_iteration=200)
+        tv_cil = TV.proximal(self.data, tau=1.0)     
```

Comment:
Could we check for a tau different from 1.0 instead of or in addition?  That would be slightly stronger, as it seems tau is currently not in the cvxpy part.

Reply:
the `tau` parameter does the same job with the regularisation parameter `alpha`. So 
```
        TV = alpha * TotalVariation(max_iteration=200)
        tv_cil = TV.proximal(self.data, tau=1.0)  
```
is the same as
```
        TV = TotalVariation(max_iteration=200)
        tv_cil = TV.proximal(self.data, tau=alpha)  
```
is the same as

```python
        TV = (alpha**2) * TotalVariation(max_iteration=200)
        tv_cil = TV.proximal(self.data, tau=1.0/alpha)  
```

Reply:
OK, so I understand that these three cases should in theory give the same result.  Please turn them into part of the unit test to make that that is also what the code achieves.

Reply:
Here I am very confused. What is the purpose of adding two more `proximal` methods  in terms of testing. `tau` is already used as `1.0`  and is just a multiplication with [the regularisation parameter](https://github.com/TomographicImaging/CIL/blob/aaa570e313e805c5b0ba90e5337365e6ca0685fa/Wrappers/Python/cil/optimisation/functions/TotalVariation.py#L204) alpha. So basically, we will run the `.proximal ` to validate that the **product** `of alpha*tau` in the following cases is the same:
- (alpha, tau) = (alpha, 1.0) is the same as
- (alpha, tau) = ( 1.0, alpha), is the same as
- (alpha, tau) = (alpha**2, 1.0/alpha).